### PR TITLE
5-get-comments-by-articles

### PR DESCRIPTION
**Pull request by Wanb00**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -131,3 +131,47 @@ describe('GET /api/articles', () => {
...
+describe('GET /api/articles/:article_id/comments', () => {
+  test('responds with all the comments of the relevant article', () => {
+    return request(app)
+    .get('/api/articles/3/comments')
+    .expect(200)
+    .then((response) => {
+      const comments = response.body.comments;
+      expect(comments.length).toBeGreaterThan(0);
+      expect(comments).toBeSortedBy('created_at', { descending: true });
+      console.log(comments)
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:144]: Remember to get rid of console.logs from finalised code 🔍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -131,3 +131,47 @@ describe('GET /api/articles', () => {
...
+      comments.forEach((comment => {
+        expect(comment).toEqual(
+          expect.objectContaining({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: 3
+          })
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:154]: Great test where it's specific where it's possible and flexible when necessary
Also great use of jest-sorted!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -131,3 +131,47 @@ describe('GET /api/articles', () => {
...
+
+  test('responds with 404 error when provided an article_id that does not exist', () => {
+    return request(app)
+    .get('/api/articles/99999/comments')
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe('Article Not Found')
+    })
+  })
+})
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:177]: Nice error handling for the errors caused by bad `article_id`s

File: __tests__/app.test.js [hunk 4]
```diff
@@ -131,3 +131,47 @@ describe('GET /api/articles', () => {
...
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: 3
+          })
+        );
+      }));
+    });
+  });
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:158]: It would be good to include another 201 happy path test that checks if there's an article that exists but has no comments, it returns a 201 and an empty array
This will require you to check if an article exists, and if it does, return an array of whatever comments belong to that article, even if it's empty!

**Review by nc-katherine-hurst — COMMENTED** [review]
Hi Wan, please find your feedback below 🙂
